api/src/main/java/org/apache/iceberg/Snapshot.java

```diff
@@ -171,4 +171,29 @@ default Iterable<DeleteFile> removedDeleteFiles(FileIO io) {
   default Integer schemaId() {
     return null;
   }
+
+  /**
+   * Return a {@link ManifestFile} for each data manifest in this snapshot which was added in
+   * addedSnapshotId
+   *
+   * @param io a {@link FileIO} instance used for reading files from storage
+   * @return a list of ManifestFile
+   */
+  default List<ManifestFile> dataManifests(FileIO io, Long addedSnapshotId) {
+    throw new UnsupportedOperationException(
+        this.getClass().getName() + " doesn't implement dataManifests(FileIO, Long)");
```

Comment:
Doesn't need to be unsupported, it can just use the existing data manifests and filer.